llvm/lib/Transforms/Jeandle/JavaOperationLower.cpp

```diff
@@ -9,65 +9,82 @@
 //===----------------------------------------------------------------------===//
 
 #include "llvm/Transforms/Jeandle/JavaOperationLower.h"
+#include "llvm/ADT/SetVector.h"
+#include "llvm/Analysis/AliasAnalysis.h"
+#include "llvm/Analysis/InlineAdvisor.h"
+#include "llvm/Analysis/InlineCost.h"
+#include "llvm/Analysis/OptimizationRemarkEmitter.h"
 #include "llvm/IR/InstIterator.h"
 #include "llvm/IR/Instructions.h"
 #include "llvm/IR/Jeandle/Attributes.h"
 #include "llvm/IR/Module.h"
-#include "llvm/Support/Debug.h"
 #include "llvm/Transforms/Utils/Cloning.h"
+#include "llvm/Transforms/Utils/ModuleUtils.h"
+#include <set>
+#include <utility>
 
 using namespace llvm;
 
 #define DEBUG_TYPE "java-operation-lower"
 
 namespace {
 
-static bool runImpl(Module &M, int Phase) {
+bool isPhaseFunc(const Function &F, int Phase) {
```

Comment:
```suggestion
bool shouldInline(const Function &F, int Phase) {
```
I think it's better to define this as a lambda expression.

Reply:
done